File: app/src/main/java/com/wavesplatform/wallet/AppBlockCanaryContext.java
package com.wavesplatform.wallet;

import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.util.Log;

import com.github.moduth.blockcanary.BlockCanaryContext;

import java.util.List;

public class AppBlockCanaryContext extends BlockCanaryContext {
    @Override
    public String provideQualifier() {
        String qualifier = "";
        try {
            PackageInfo info = App.getAppContext().getPackageManager()
                    .getPackageInfo(App.getAppContext().getPackageName(), 0);
            qualifier += info.versionCode + "_" + info.versionName + "_YYB";
        } catch (PackageManager.NameNotFoundException e) {
            Log.e("AppBlockCanaryContext", "provideQualifier exception", e);
        }
        return qualifier;
    }

    @Override
    public String provideUid() {
        return "87224330";
    }

    @Override
    public String provideNetworkType() {
        return "wifi";
    }

    @Override
    public int provideMonitorDuration() {
        return 9999;
    }

    @Override
    public int provideBlockThreshold() {
        return 500;
    }

    @Override
    public boolean displayNotification() {
        return BuildConfig.DEBUG;
    }

    @Override
    public List<String> concernPackages() {
        List<String> list = super.provideWhiteList();
        list.add("com.wavesplatform");
        return list;
    }

    @Override
    public List<String> provideWhiteList() {
        List<String> list = super.provideWhiteList();
        list.add("com.whitelist");
        return list;
    }

    @Override
    public boolean stopWhenDebugging() {
        return true;
    }
}

File: app/src/main/java/com/wavesplatform/wallet/AppBlockCanaryContext.kt
/*
 * Created by Eduard Zaydel on 1/4/2019
 * Copyright © 2019 Waves Platform. All rights reserved.
 */

package com.wavesplatform.wallet

import com.github.moduth.blockcanary.BlockCanaryContext

class AppBlockCanaryContext : BlockCanaryContext() {
    override fun provideQualifier(): String {
        return BuildConfig.VERSION_CODE.toString() + "_" + BuildConfig.VERSION_NAME + "_YYB"
    }

    override fun provideNetworkType(): String {
        return "wifi"
    }

    override fun displayNotification(): Boolean {
        return BuildConfig.DEBUG
    }

    override fun stopWhenDebugging(): Boolean {
        return true
    }
}